index.js

```diff
@@ -0,0 +1,241 @@
+            } else {
+                position.position_mappings.forEach(async (mapping) => {
+                    try {
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Crew");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.add(roleVar);
+                        }
+                    } catch (error) {
+                        console.error("an error occured while editing roles:", error);
+                    }
+                })
+            }
+        });
+    });
+};
+
+
+//when the bot turns on
+phoenixClient.on('ready', () => {
+    console.log(`Logged in as ${phoenixClient.user.tag}!`);
+    updateRoles();
+    setInterval(function () {
+        updateRoles();
+        console.log("half an hour has gone by, updating roles");
+    }, 1800000);
+});
+
+//when someone joins the server
+phoenixClient.on('guildMemberAdd', member => {
+    updateRoles();
+})
+
+//when there is a new message in the server
+phoenixClient.on('messageCreate', async (message) => {
+    if (message.author.bot) return;
+    if (message.content.includes("chief")) {
+        message.channel.send("det heter gruppeleder! :rage:");
+    };
+    const prefix = '!';
+    //check if message is a command
+    if (message.content.startsWith(prefix)) {
+        const args = message.content.slice(prefix.length).trim().split(/ +/);
+        const command = args.shift().toLowerCase();
+        //handle commands
+        switch (command) {
+            case 'help':
+                const helpEmbed = new EmbedBuilder()
+                    .setColor(0x0099FF)
+                    .setTitle('Phoenix bot kommandoer')
+                    .setDescription('Kommandoer du kan bruke')
+                    .addFields({
+                        name: '!help',
+                        value: 'Denne kommandoen. viser hvilke kommandoer du kan bruke'
+                    }, {
+                        name: '!roles',
+                        value: 'Oppdaterer roller.',
+                    }, {
+                        name: '!liam',
+                        value: 'Viser den nåværende tiden i Japan',
+                    })
+                    .setTimestamp()
+                    .setFooter({
+                        text: 'Phoenix bot !help',
+                    });
+                message.reply({
+                    embeds: [helpEmbed]
+                });
+                break;
+            case 'roles':
+                //only runs if user has role "Administrasjon" in the discord server
+                const guild = phoenixClient.guilds.cache.get(phoenixGuildId);
+                let role = message.member.roles.cache.find(role => role.name === "Administrasjon")
```

Comment:
There's no need for this variable to be mutable. Additionally, the role should be fetched by its id, rather than the name.
```suggestion
                const role = message.member.roles.cache.find(role => role.name === "Administrasjon")
```

Reply:
nei.